integration-tests/main_test.go

```diff
@@ -0,0 +1,211 @@
+package integrationtests
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/docker/go-connections/nat"
+
+	"github.com/hashicorp/consul/api"
+)
+
+// syntheticNodeName is the name given to the "synthetic" node services are
+// registered to.
+const syntheticNodeName = "synthetic-node"
+
+var (
+	// serverImage is the container image reference for the Consul server. It can
+	// be configured using the -server-image flag.
+	serverImage string
+
+	// dataplaneImage is the container image reference for Consul Dataplane. It
+	// can be configured using the -dataplane-image flag.
+	dataplaneImage string
+
+	// outputDir is the directory artifacts will be written to. It can be configured
+	// using the -output-dir flag.
+	outputDir string
+
+	// disableReaper controls whether the container reaper is enabled. It can be
+	// configured using the -disable-reaper flag.
+	disableReaper bool
+
+	// upstreamLocalBindPort is the port the frontend sidecar will bind the local
+	// listener for the backend upstream to.
+	upstreamLocalBindPort = tcpPort(10000)
+
+	// proxyInboundListenerPort is the port the backend sidecar will bind its
```

Comment:
looks like both the frontend and backend sidecars have the same public listener port (not just backend per comment).

Reply:
Yep, you're totally right, will update! It's just that we don't ever actually use the frontend proxy's public listener port.